Deprecated client.Apply used in littlered_controller.go

```
 Error: internal/controller/littlered_controller.go:1303:27: SA1019: client.Apply is deprecated: Use client.Client.Apply() and client.Client.SubResource("subrsource").Apply() instead. (staticcheck)
  	return r.Patch(ctx, obj, client.Apply, client.FieldOwner(fieldManager), client.ForceOwnership)
```

This issue is currently ignored by linting due to a comment.

--- a/internal/controller/littlered_controller.go
+++ b/internal/controller/littlered_controller.go
@@ -31,6 +31,7 @@ import (
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/util/retry"
@@ -1300,7 +1301,12 @@ func (r *LittleRedReconciler) apply(ctx context.Context, owner *littleredv1alpha
 		return err
 	}
 	obj.GetObjectKind().SetGroupVersionKind(gvk)
-	return r.Patch(ctx, obj, client.Apply, client.FieldOwner(fieldManager), client.ForceOwnership) //nolint:staticcheck
+	raw, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return err
+	}
+	u := &unstructured.Unstructured{Object: raw}
+	return r.Apply(ctx, client.ApplyConfigurationFromUnstructured(u), client.FieldOwner(fieldManager), client.ForceOwnership)
 }
 
 // getRedisPassword retrieves the Redis password from the secret if auth is enabled

